frontend/src/components/RisLawPreview.vue

```diff
@@ -264,18 +284,100 @@ watch(
   },
   { deep: true },
 )
+
+/* -------------------------------------------------- *
+ * Focus handling                                     *
+ * -------------------------------------------------- */
+
+// NB: If we need this in more places in the future, consider extracting
+// this into a composable to reuse it in other places!
```

Comment:
i think i would have already moved this into a composable, in my experience that also makes testing a bit simpler ^^

Reply:
I could not really decide on the right abstraction here. Thought it would be easier once we have more than one use case 😅